### ISSUE TITLE
Add support to client certificates

### ISSUE THREAD

**Issue by ogarcia** [russdill/juniper-vpn-py#1]
Hi,

If after that line https://github.com/russdill/juniper-vpn-py/blob/master/juniper-vpn.py#L117 you add this code:

``` python
self.br.add_client_certificate('https://'
  + self.args.host,"/path/to/client/key.pem","/path/to/client/cert.pem")
```

Your python then add suppor to client certificates. Only need put pretty to set as options.

Note: Is important remove password in client key.

Furthermore, I think that do this:

``` python
while True:
  action = self.next_action()
  if action == 'tncc':
    self.action_tncc()
  elif action == 'login':
    self.action_login()
  elif action == 'key':
    self.action_key()
  elif action == 'continue':
    self.action_continue()
  elif action == 'connect':
    self.action_connect()
```

Is a problem because if your connection breaks in any moment after some hours connected, your cookie is invalid and cannot connect again, and the process is always trying reconnect. I add a `break` after `self.action_connect()` and use _systemd_ to control process.

**Comment by themiurgo** [comment]
:+1: works for me too.

**Comment by xortim** [comment]
Client cert support would definitely be nice

**Comment by russdill** [comment]
I've noticed a very different kind of client certificate requirement and merged support in 86e16b21f8ecb60db55aa60039cf580b36e1cec3

**Comment by cardonator** [comment]
@russdill I still seem to be having an issue. My juniper host checks for a client certificate after login. Even passing them such as

TNCC_FUNK=1 TNCC_CERTS=key.crt.pem,key.key.pem openconnect --juniper --user myuser --csd-wrapper tncc.py juniper.example.com

after putting my passwords in I get

SSL negotiation with juniper.example.com
Connected to HTTPS on juniper.example.com
Failed to read from SSL socket: Rehandshake was requested by the peer.
Error fetching HTTPS response
Failed to obtain WebVPN cookie
